uwblueprint/community-fridge-kw
# feat: create forget password frontend and backend workflow

## Brief description. What is this change? 

### [Forget Password Flow](https://www.notion.so/uwblueprintexecs/Forget-Password-Flow-752d773c0f0d41dea09d9a13bc5f6b89)

This implements the frontend and backend flow of the forget password screens. There are two entry points - either in the login screen or inside of the "My Account" screen for a logged in user. 

See video demo: 

https://user-images.githubusercontent.com/19617248/153780045-70e5126c-cd2b-49f9-89c8-4c329043eb61.mp4




## Implementation description. How did you make this change?
* Leveraged the existing starter code reset password but with a custom email handler: 
* Go through the [figma forget password frames](https://www.figma.com/file/FAiUthK6jklFWCXwvXaitd/Donor-Experience-W22?node-id=233%3A10223) and make calls to the respective services 
 1) Send Password Reset Link: https://github.com/uwblueprint/community-fridge-kw/blob/9f29527455f4c06912d07d0d8e6630ca4ce85a70/backend/typescript/services/implementations/authService.ts#L124
 2) [Firebase verify password reset code](https://firebase.google.com/docs/reference/rest/auth/#section-verify-password-reset-code)
 3) [Firebase confirm password reset change](https://firebase.google.com/docs/reference/rest/auth/#section-confirm-reset-password)



## Steps to test
1. To test you'll have to change the firebase action url to the one from this PR or to localhost (if you're running it locally) 
![image](https://user-images.githubusercontent.com/19617248/153781804-586931fd-198c-4225-bd7e-2b61aa74f221.png)
2. Test out the forget password steps and verify you get an email and are able to log in with your new password.



## Checklist
